tests/common/UrlGenerators.cpp

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright (C) 2020 HERE Europe B.V.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * License-Filename: LICENSE
+ */
+
+#include "UrlGenerators.h"
+
+#include <string>
+#include <vector>
+
+// clang-format off
+#include "generated/parser/ApiParser.h"
+// clang-format on
+#include <olp/core/generated/parser/JsonParser.h>
+#include <olp/dataservice/read/model/Partitions.h>
+#include <olp/dataservice/read/model/VersionResponse.h>
+#include "generated/model/Api.h"
+
+std::string PathGenerator::GetPartitions(
+    const std::string& layer,
+    const olp::dataservice::read::PartitionsRequest::PartitionIds& partitions,
+    uint64_t version) {
+  std::string path = "/layers/" + layer + "/partitions?";
+  for (const auto& partition : partitions) {
+    path.append("partition=" + partition + "&");
+  }
+  path.append("version=" + std::to_string(version));
+  return path;
+}
+
+std::string PathGenerator::GetData(const std::string& layer,
+                                   const std::string& data_handle) {
+  return "/layers/" + layer + "/data/" + data_handle;
+}
+
+std::string PathGenerator::GetLatestVersion() {
+  return "/versions/latest?startVersion=-1";
+}
+
+std::string PathGenerator::GetQuadKey(const std::string& quadkey,
+                                      const std::string& layer,
+                                      uint64_t version, uint64_t depth) {
+  return "/layers/" + layer + "/versions/" + std::to_string(version) +
+         "/quadkeys/" + quadkey + "/depths/" + std::to_string(depth);
+}
+
+std::string PathGenerator::FullPath(const std::string& apis_response,
```

Comment:
maybe we can use apis model instead, so no need for converting apis to json and back.

Reply:
In this case we will have dependancy in integration tests fron src. Here we want to keep all such dependancies in test-common lib.